de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ArchiveIncomingReferencePointNegotiation.java

```diff
@@ -26,24 +25,26 @@
 import de.fu_berlin.inf.dpp.net.xmpp.JID;
 import de.fu_berlin.inf.dpp.net.xmpp.XMPPConnectionService;
 import de.fu_berlin.inf.dpp.observables.FileReplacementInProgressObservable;
+import de.fu_berlin.inf.dpp.session.IReferencePointManager;
 import de.fu_berlin.inf.dpp.session.ISarosSession;
 import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
 import de.fu_berlin.inf.dpp.util.CoreUtils;
 
 /**
- * Implementation of {@link AbstractIncomingProjectNegotiation} utilizing a
- * transferred zip archive to exchange differences in the project files.
+ * Implementation of {@link AbstractIncomingReferencePointNegotiation} utilizing
+ * a transferred zip archive to exchange differences in the files from reference
+ * points
  */
-public class ArchiveIncomingProjectNegotiation extends
-    AbstractIncomingProjectNegotiation {
+public class ArchiveIncomingReferencePointNegotiation extends
+    AbstractIncomingReferencePointNegotiation {
 
     private static final Logger LOG = Logger
-        .getLogger(ArchiveIncomingProjectNegotiation.class);
+        .getLogger(ArchiveIncomingReferencePointNegotiation.class);
 
-    public ArchiveIncomingProjectNegotiation(
+    public ArchiveIncomingReferencePointNegotiation(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Constructor has 11 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS00107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=273&issues=AWaWrfbgsSPV6ZXI_B8_&open=AWaWrfbgsSPV6ZXI_B8_)

[ISSUE_KEY]: AWaWrfbgsSPV6ZXI_B8_